cpp/src/arrow/array/array_run_end_test.cc

```diff
@@ -543,8 +540,9 @@ TEST_P(TestRunEndEncodedArray, Validate) {
     malformed_array->data()->buffers.emplace_back(NULLPTR);
     EXPECT_RAISES_WITH_MESSAGE_THAT(
         Invalid,
-        ::testing::HasSubstr("Invalid: Values array invalid: Expected 2 buffers in array "
-                             "of type int32, got 3"),
+        ::testing::HasSubstr(
+            "Invalid: Values array invalid: Invalid: Expected 2 buffers in array "
+            "of type int32, got 3"),
```

Comment:
Isn't a bit weird to have "Invalid: " twice in the message?

Reply:
It is, but this is the format of other child array validation failures (since the parent array's error message contains the child array's). Possibly a follow up issue is in order to simplify those; I think it'd be most valuable to provide a field path to the child which failed validation and the "leaf" error message.